lib/utils/manager/storage_directory_manager.dart

```diff
@@ -13,7 +13,8 @@ class StorageDirectoryManager {
 
   static StorageDirectoryManager get instance => _instance;
 
-  Future<String> getFileStoreDirectory() async {
+  Future<String> getFileStoreDirectory({bool isTemporary = true}) async {
+    if (!isTemporary) return (await getDownloadsDirectory())!.path;
```

Comment:
can you explain why you need to have `isTemporary` here

Reply:
I had to do this because without that on desktop the downloaded files by the users are stored in the temporary directory (for example if I want to download a .pdf shared in a room). 

This solution allow the app to save those files in  `download_directory/Twake Chat/`, same as Signal for example.

Other files are still saved in the `/tmp` folder which is the cache folder

Reply:
Demo on desktop:

[Capture vidéo du 14-05-2024 12:26:11.webm](https://github.com/linagora/twake-on-matrix/assets/31937920/cda8c13c-6056-4d4e-9235-b697e01c5ed2)

Reply:
I agree there must be a cleaner way to do this, since it works but I propose to enhance this logic in an other pull request. Because this refactoring means a lot of other changes

Reply:
make an ADR to improve later

Reply:
ticket created: https://github.com/linagora/twake-on-matrix/issues/1767